libyul/optimiser/Metrics.h

```diff
@@ -71,16 +75,21 @@ class CodeSize: public ASTWalker
 class CodeCost: public ASTWalker
 {
 public:
-	static size_t codeCost(Expression const& _expression);
+	static size_t codeCost(Dialect const& _dialect, Expression const& _expression);
 
 private:
+	CodeCost(Dialect const& _dialect): m_dialect(_dialect) {}
+
 	void operator()(FunctionCall const& _funCall) override;
 	void operator()(FunctionalInstruction const& _instr) override;
 	void operator()(Literal const& _literal) override;
 	void visit(Statement const& _statement) override;
 	void visit(Expression const& _expression) override;
 
 private:
+	void instructionCost(dev::eth::Instruction _instruction);
```

Comment:
A bit pedantic, but the name sounds like it returns the instruction cost... maybe ``addInstructionCost``? Or just ``operator()(dev::eth::Instruction)``?